Automattic/wp-calypso
# POC: Make /plugins section available to Logged Out users

#### Proposed Changes

More info: pau2Xa-4hD-p2

This adds the changes needed to get the /plugins section to be rendered isomorphically and accessible to logged-out users.
Changes include:
- updated the sections configs to enable isomorphically and logged-out use of the /plugins 
- neutralized client-only code
- prefetched data inside route middleware so they're ready for when the components are rendered

#### Testing Instructions


* make sure you're logged out 
* disable JavaScript
* visit these links:

##### /plugins
<img width="480" alt="Screenshot on 2022-07-29 at 17-02-47" src="https://user-images.githubusercontent.com/2749938/181777489-e8fb9fd5-4b27-4924-b8a5-fb336eca48f9.png">

##### /plugins?s=seo
<img width="480" alt="Screenshot on 2022-07-29 at 17-03-29" src="https://user-images.githubusercontent.com/2749938/181777541-6a345425-c021-4269-bc78-bd1ad6e6294e.png">

##### /plugins/browse/seo
<img width="480" alt="Screenshot on 2022-07-29 at 17-03-12" src="https://user-images.githubusercontent.com/2749938/181777520-59ecd157-e0b8-4fa9-bec5-74de469fd860.png">

##### /plugins/wordpress-seo-premium
<img width="480" alt="Screenshot on 2022-07-29 at 17-04-01" src="https://user-images.githubusercontent.com/2749938/181777579-343f68ed-15db-4b76-977b-7d791ae00f10.png">

TBD

#### TODOs
- [ ] fix header breadcrumbs
- [ ] add a relevant CTA on the Plugin details page
- [ ] optimize and avoid unnecessary prefetches by following the component conditions
- [ ] improve `renderToString` tests by passing in mock data
- [ ] address any other failing e2e and unit tests
- [ ] make SeachBox keywords and category link clickable 
- [ ] address suppressed Plugin screenshots parsing
- [ ] restore `useLocalizeUrl` functionality
- [ ] integrate `controllers-2` into the main one
- [ ] ensure localization still works on the server-rendered version

#### Pre-merge Checklist






Related to #
